Rendera/include/cube.hpp

```diff
@@ -0,0 +1,155 @@
+#ifndef CUBE_HPP
+#define CUBE_HPP
+
+#include <limits>
+#include "vec3.hpp"
+#include "ray.hpp"
+#include "material.hpp"
+
+
+vec3 max(vec3 a, vec3 b, ray r)
+{
+    if (dot((a - r.get_origin()),r.get_direction()) > dot((b- r.get_origin()),r.get_direction()))
+    {
+        return a;
+    }
+    else
+    {
+        return b;
+    }
+}
+
+vec3 min(vec3 a, vec3 b, ray r)
+{
+
+    if (dot((a - r.get_origin()),r.get_direction()) < dot((b- r.get_origin()),r.get_direction()))
+    {
+        return a;
+    }
+    else
+    {
+        return b;
+    }
+}
+
+class Cube
+{
+private:
+    vec3 m_position;
+    vec3 m_a, m_b, m_c;
+    Material m_material;
+
+public:
+    Cube(vec3 position, vec3 a, vec3 b, vec3 c, Material mat) : m_position{position}, m_a{a}, m_b{b}, m_c{c}, m_material{mat} {};
+
+    ray normal(const vec3 &point)
+    {
+        vec3 p_pos, np_pos;
```

Comment:
In modern programming, there is no advantage in declaring without assigning. You should write `vec3 p_pos = point - m_position;` and `vec3 np_pos = ...`.

Reply:
Also, what are `p_pos` and `np_pos`?

Reply:
p_pos : vector from m_position to point
np_pos : vector from diagonally opposite corner to point